src/containers/Charts/Line.tsx

```diff
@@ -0,0 +1,253 @@
+import Select from "@material-ui/core/Select";
+import Button from '@material-ui/core/Button';
+
+import { Blockchain, Block, State } from "../../types";
+import {
+    convertTimeStamp,
+    getSelectedDate
+} from "./helpers";
+
+const mapState = (state: State): Blockchain => ({
+    blokchain: state.tezos.blocks
+});
+
+const LineCharts = (): React.ReactElement => {
+    const dispatch = useDispatch();
+    const { blokchain } = useMappedState(mapState);
+    const [dateFrom, setDateFrom] = useState(getSelectedDate(7));
+    const [label, setLabel] = useState([
+        "19-04-2019",
+        "20-04-2019",
+        "21-04-2019",
+        "22-04-2019"
+    ]);
+    const [data, setData] = useState([10, 20, 30, 40]);
+    const [config, setConfig] = useState({
+        chartType: "buyers",
+        label: "Amount of the currency",
+        title: "Transactions for the selected buyer and day"
+    });
+    const [select, setSelect] = useState("buyer");
+    const [buyer, setBuyer] = useState("buyer");
+    const [seller, setSeller] = useState("xyz2");
+    const [chartType, setChartType] = useState("buyer");
+
+    const handleBuyerChange = (e: any) => {
+        setTimeout(() => setBuyer(e.target.value), 100)
+    };
+
+    const changeUserType = (e: any) => {
+        setChartType(e.target.value);
+    }
+
+    const filterChart = (blokchain: Block[], chartType: string): void => {
+        const labels: any[] = [];
+        let elements: number[] = [];
+
+        blokchain.forEach((item: Block): void => {
+            const timeStampConverted: string = convertTimeStamp(item.timestamp);
+
+            if (timeStampConverted === dateFrom) {
+                switch (chartType) {
+                    case "selers":
+                        if (item.source == seller) {
+                            elements.push(item.amount);
+                        }
+
+                        break;
+                    case "buyers":
+                        if (item.destination == buyer) {
+                            elements.push(item.amount);
+                        }
+
+                        break;
+                    default:
+                }
+            }
+        });
+        elements.forEach(() => {
+            labels.push(dateFrom);
+        });
+
+        setLabel(labels.slice(0, 100));
+        setData(elements.slice(0, 100));
+        setLoaderFalse();
+
+    };
+
+    const triggerSetDateFrom = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        // setLoaderTrue();
+        setDateFrom(e.target.value);
+    };
+
+    const setLoaderFalse = (): void => {
+        dispatch({
+            type: "LOADER_STATE",
+            show: false
+        });
+    };
+
+    const setLoaderTrue = (): void => {
+        dispatch({
+            type: "LOADER_STATE",
+            show: true
+        });
+    };
+
+    const submitChart = () => {
+        setLoaderTrue();
+        setTimeout(() => filterChart(blokchain, config.chartType), 100);
```

Comment:
to samo co prędzej - co to za timeout?

Reply:
Tutaj tego nie możemy rozgryźć, on jest potrzebny żeby działał loader z reduxa i wczytywał wykres.

Reply:
Po usunieciu setTimeout u mnie wszystko dziala. Na pewno jest potrzebny?